php/Terminus/Collections/Sites.php

```diff
@@ -82,32 +84,35 @@ public function fetch(array $arg_options = [])
         'org_id'    => null,
         'team_only' => false,
         ];
-        $options         = array_merge($default_options, $arg_options);
+        $options = array_merge($default_options, $arg_options);
 
         $sites = [];
         if (is_null($options['org_id'])) {
-            $sites = $this->user->getSites();
+            $sites[] = $this->user->getSites();
         }
         if (!$options['team_only']) {
             $memberships = $this->user->org_memberships->fetch()->all();
+            if (!is_null($org_id = $options['org_id']) && ($org_id != 'all')) {
+                $memberships = [$memberships[$org_id],];
+            }
             foreach ($memberships as $membership) {
                 if ($membership->get('role') != 'unprivileged') {
-                    $sites = array_merge($sites, $membership->organization->getSites());
+                    $sites[] = $membership->organization->getSites();
                 }
             }
         }
-
-        foreach ($sites as $site) {
-            if (!isset($this->models[$site->id])) {
-                $site->collection        = $this;
-                $this->models[$site->id] = $site;
-            } else {
-                $this->models[$site->id]->memberships[] = array_merge(
-                    $this->models[$site->id]->memberships,
-                    $site->memberships
-                );
+        $merged_sites = [];
+        foreach ($sites as $site_group) {
+            foreach ($site_group as $site) {
+                if (!isset($merged_sites[$site->id])) {
+                    $merged_sites[$site->id] = $site;
+                } else {
+                    $merged_sites[$site->id]->memberships[] = $site->memberships[0];
+                }
             }
         }
+        $this->models = $merged_sites;
```

Comment:
This instinctually feels better, no changing models inside loops. Other than my gut likes it, are there potential performance, or other, gains doing it this way?